ecl/eclcc/NEWACTIVITY.rst

```diff
@@ -0,0 +1,537 @@
+
+  Again this is always required, so the list of fields should follow the number of partitions.
+
+* Which fields are returned?
+
+  Normally the input row, but often it would be useful for the output to include details of which quantile a row
+  corresponds to.  To allow this an optional transform should be that could be passed the input row as LEFT and
+  the quantile number as COUNTER.
+
+* How about first and last rows in the dataset?
+
+  Sometimes it is also useful to know the first and last rows.  Add flags to allow them to be optionally returned.
+
+* How do you cope with too few input rows (including an empty input)?
+
+  After some discussion we decided that QUANTILE should always return the number of parts requested.  If there were fewer
+  items in the input they would be duplicated as appropriate.  We should provide a DEDUP flag for the situations when that
+  is not desired.
+  If there is an empty dataset as input then the default (blank) row will be created.
+
+* Should all rows have the same weighting?
+
+  Generally you want the same weighting for each row.  However, if you are using QUANTILE to split your dataset, and the cost
+  of the next operation depends on some feature of the row (e.g., the frequency of the firstname) then you may want to
+  weight the rows differently.
+
+* What if we are only interested in the 5th and 95th centiles?
+
+  We could optionally allow a set of values to be selected from the results.
+
+There were also some implementation details that came out of the discussions:
+
+* How accurate should the results be?
+
+  The simplest implementation of QUANTILE (sort and then select the correct rows) will always produce accurate
+  results.  However, there may be some implementations that can produce an approximate answer more quickly.  Therefore
+  we could add a SKEW attribute to allow early termination.
+
+* Does the implementation need to be stable?
+
+  In other words, if there are rows with identical values for the ordering fields, but other fields not part of the
+  ordering with different values, does it matter which of those rows are returned?  Does the relative order within those
+  matching rows matter?
+  
+  The general principle in the HPCC system is that sort operations should be stable, and that where possible activities
+  return consistent, reproducible results.  However, that often has cost - either in performance or memory consumption.
+  The design discussion highlighted that if all the fields from the row are included in the sort order then
+  the relative order does not matter because the duplicate rows will be indistinguishable.  (This is also true for
+  sorts, and following the discussion an optimization was added to 5.2 to take advantage of it.)  For the QUANTILE
+  activity we will add an ECL flag,  but also aim to spot it automatically.
+  
+* Returning counts of the numbers in each quantile might be interesting.
```

Comment:
"Returning counts of the numbers in each quantile might be interesting.".What have I grossly misunderstood? Quantiles, themselves, are the values at the boundaries of the partition, but the partitions themselves have n/q (n = size of dataset & q = number of partitions e.g. median => n = 2) values by definition. So what have I missed or what did you mean by "counts of the numbers in each quantile"?